app/db.py

```diff
@@ -0,0 +1,26 @@
+from datetime import timedelta
+
+
+def get_start_date(conn, symbol):
+    cursor = conn.cursor()
+    try:
+        cursor.execute("SELECT MAX(date) FROM ohlcv WHERE symbol = %s", (symbol,))
+        start_date, = cursor.fetchone() # returns none if there are no records
+
+        if start_date:
+            start_date += timedelta(days=1)
+    finally:
+    	cursor.close()
```

Comment:
I know I wrote the cursor code originally, but I was doing it in a hurry and I did a bad thing.  I used `try` and `finally` to be ensure the cursor would get closed, but the right way to do it is with the [`with` statement](https://www.python.org/dev/peps/pep-0343/).  Something like:

```
with conn.cursor() as cursor:
    cursor.execute("SELECT foo, bar FROM some_table")
    foos_and_bars = cursor.fetchall()

for foo, bar in foos_and_bars:
    print foo, bar
```

This will automatically close the cursor after the `fetchall` line and before the `for` loop.  Same thing (using `with`) applies to the cursor in `save_stock_data` and I think all uses of cursors I've ever seen.